sdk/react/src/DVCProvider.tsx

```diff
@@ -1,24 +1,36 @@
 import { ProviderConfig } from './types'
-import React, { ReactNode, useEffect, useMemo, useState } from 'react'
+import React, { ReactNode, useEffect, useState } from 'react'
 import initializeDVCClient from './initializeDVCClient'
 import { Provider } from './context'
+import { DVCClient } from '@devcycle/devcycle-js-sdk'
 
 type Props = {
   config: ProviderConfig
   children: ReactNode
 }
 
+let client: DVCClient
+
 export default function DVCProvider(props: Props): React.ReactElement {
     const { envKey, user, options } = props.config
-    const client = useMemo(() => initializeDVCClient(envKey, user, options), [envKey, user, options])
+
+    if (!client || client.closing) {
```

Comment:
what this is trying to do is basically say "if we've already created a client via this component but its been unmounted (and therefore that client was closed) then we need a new client since this component has been created again.